### PR TITLE
Fix Some Issues, Improve Code quality

### PR DESCRIPTION
**Pull request by Kissaki**
Fixing some stuff I found with PVS-Studio.

I also noticed: In `src/mumble` we have EnvUtils to handle getting environment variables. This should be moved to `src`, so the code in `src/murmur` can also make use of it (there’s another place I didn't touch as well).

CryptState also still has two fields which I don’t know how to correctly initialize (openssh types), so for now I didn’t (I believe these missing init. were pointed to by cppcheck as well).

### CODE REVIEW

File: src/mumble/EnvUtils.cpp [hunk 1]
```diff
@@ -17,14 +17,20 @@ QString EnvUtils::getenv(QString name) {
 	const wchar_t *wname = reinterpret_cast<const wchar_t *>(name.utf16());
 
 	// Query the required buffer size (in elements).
-	_wgetenv_s(&requiredSize, 0, 0, wname);
+	errno_t res = _wgetenv_s(&requiredSize, 0, 0, wname);
+	if (res != 0) {
+		QString();
```
Comment (davidebeatrici) [inline src/mumble/EnvUtils.cpp]: What does this do?

Comment (davidebeatrici) [inline src/mumble/EnvUtils.cpp]: Shouldn't this be `return QString()`?

File: src/mumble/EnvUtils.cpp [hunk 2]
```diff
@@ -17,14 +17,20 @@ QString EnvUtils::getenv(QString name) {
...
 	if (requiredSize == 0) {
 		return QString();
 	}
 
 	// Resize buf to fit the value and put it there.
 	buf.resize(static_cast<int>(requiredSize * sizeof(wchar_t)));
-	_wgetenv_s(&requiredSize, reinterpret_cast<wchar_t *>(buf.data()), requiredSize, wname);
+	errno_t res = _wgetenv_s(&requiredSize, reinterpret_cast<wchar_t *>(buf.data()), requiredSize, wname);
+	if (res != 0) {
+		QString();
```
Comment (davidebeatrici) [inline src/mumble/EnvUtils.cpp]: As above.

File: src/mumble/EnvUtils.cpp (hunk 1 above)
Comment (Kissaki) [inline src/mumble/EnvUtils.cpp]: Yes, fixed. I also dropped the res variable.

File: src/mumble/EnvUtils.cpp (hunk 2 above)
Comment (Kissaki) [inline src/mumble/EnvUtils.cpp]: Yes, fixed. I also dropped the res variable.

File: src/murmur/main.cpp [hunk 3]
```diff
@@ -205,16 +205,25 @@ int main(int argc, char **argv) {
...
 #ifdef Q_OS_WIN
+	// By default, windbus expects the path to dbus-daemon to be in PATH, and the path
+	// should contain bin\\, and the path to the config is hardcoded as ..\etc
+
 	{
 		size_t reqSize;
-		_wgetenv_s(&reqSize, NULL, 0, L"PATH");
-		if (reqSize > 0) {
+		if (_wgetenv_s(&reqSize, NULL, 0, L"PATH") != 0) {
+			qWarn() << "Failed to get PATH. Not adding application directory to PATH. DBus bindings may not work.";
```
Comment (Kissaki) [inline src/murmur/main.cpp]: Missing include

Comment (Kissaki) [inline src/murmur/main.cpp]: Actually not include, it's qWarning(), not qWarn() :)

File: src/mumble/VoiceRecorder.cpp [hunk 4]
```diff
@@ -294,7 +294,7 @@ bool VoiceRecorder::ensureFileIsOpenedFor(SF_INFO& soundFileInfo, boost::shared_
 void VoiceRecorder::run() {
 	Q_ASSERT(!m_recording);
 	
-	if (g.sh && g.sh->uiVersion < 0201003)
```
Comment (mkrautz) [inline src/mumble/VoiceRecorder.cpp]: WTF? Why did we do this in the first place?!

File: src/mumble/WASAPI.cpp [hunk 5]
```diff
@@ -1065,7 +1065,9 @@ void WASAPIOutput::run() {
 	if (pwfx)
 		CoTaskMemFree(pwfx);
 
-	setVolumes(pDevice, false);
+	if (pDevice) {
+		setVolumes(pDevice, false);
```
Comment (mkrautz) [inline src/mumble/WASAPI.cpp]: setVolumes is called in other contexts where you don't do the null check.
Maybe it should be moved into the function itself?

Comment (Kissaki) [inline src/mumble/WASAPI.cpp]: It’s only being called in `WASAPIOutput::run()` (3 times), after the initial pDevice null check which jumps to cleanup.

**Pull request merged**